packages/ai/src/Button.ts

```diff
@@ -58,13 +87,185 @@ class Button extends UI5Element {
 	@property({ type: Boolean })
 	disabled!: boolean;
 
+	/**
+	 * Defines the current state of the component.
+	 *
+	 * **Note:** if nothing is defined, the component will be set initially to the first defined state (if any).
+	 *
+	 * @default ""
+	 * @public
+	 */
 	@property()
-	icon!: string;
+	state?: string;
 
-	@property()
-	endIcon!: string;
+	/**
+	 * Keeps the current state of the component.
+	 * @private
+	 */
+	@property({ type: Object })
+	_currentStateObject?: ButtonState;
+
+	/**
+	 * Initiates button elements fade-out phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeOut!: boolean;
+
+	/**
+	 * Initiates button fade middle phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeMid!: boolean;
+
+	/**
+	 * Initiates button elements fade-in phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeIn!: boolean;
+
+	/**
+	 * Defines the available states of the component.
+	 * **Note:** Although this slot accepts HTML Elements, it is strongly recommended that you only use `ui5-ai-button-state` components in order to preserve the intended design.
+	 * @public
+	 */
+	@slot({ type: ButtonState, "default": true })
+	states!: Array<ButtonState>;
+
+	onBeforeRendering(): void {
+		if (this.fadeOut || this.fadeIn) {
+			return;
+		}
+
+		if (this.state && this._currentStateObject && !Object.keys(this._currentStateObject).length) {
```

Comment:
!Object.keys... Do we verify that there is an object? Can we just use !this._currentStateObject or check any of the values?

Reply:
even if it is not defined, it is empty object {}, so this._currentStateObject will be always truthy.